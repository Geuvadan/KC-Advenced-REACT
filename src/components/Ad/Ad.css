.button {
  margin-top: 0.5rem;
  width: 100%;
  position: relative;
  bottom: 0px;
  display: block;
  line-height: 2rem;
  font-size: 0.9rem;
  cursor: pointer;
  background-color: lightgrey;
  text-align: center;
  text-decoration: none;
  color: black;
}

.ads-grid {
  margin-top: 20px;
  display: grid;
  grid-template-columns: 1fr;
  grid-gap: 3rem;
}

.ad {
  box-shadow: 5px 5px 10px 5px rgba(0, 0, 0, 0.5);
  padding: 1rem;
}

@media all and (min-width: 768px) {
  .ads-grid {
    grid-template-columns: 1fr 1fr;
  }
}

@media all and (min-width: 992px) {
  .ads-grid {
    grid-template-columns: 1fr 1fr 1fr;
  }
}
