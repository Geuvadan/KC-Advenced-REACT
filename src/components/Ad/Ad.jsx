import React from 'react';

export default class Ad extends React.Component {
  state = {
    ads: this.props.data,
  };

  ad = this.state.ads.map((item) => {
    return (
      <div key={item._id}>
        <p>{item.name}</p>
      </div>
    );
  });

  render() {
    return <div>{this.ad}</div>;
  }
}
