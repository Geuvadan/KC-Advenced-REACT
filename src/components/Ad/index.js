import { connect } from 'react-redux';

import Ad from './Ad';

function mapStateToProps(state, ownProps) {
  return {
    ads: state.ads.results,
  };
}

export default connect(mapStateToProps)(Ad);
