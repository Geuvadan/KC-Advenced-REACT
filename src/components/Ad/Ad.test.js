import React from 'react';
import { shallow } from 'enzyme';
import Ad from './Ad';

describe('Ad', () => {
  const props = {
    ads: [
      {
        _id: '1234567890',
        name: 'test_Ad_1',
        type: 'sell',
        price: 100,
      },
      {
        _id: '1234567890',
        name: 'test_Ad_1',
        type: 'sell',
        price: 100,
      },
    ],
  };

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Ad {...props} />);
  });

  test('Snapshot testing', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Shoud render', () => {
    expect(wrapper.exists()).toBe(true);
  });

  test('Shoud render an item with class ads-grid', () => {
    expect(wrapper.hasClass('ads-grid')).toBe(true);
  });

  test('Snapshot testing', () => {
    expect(wrapper.find('.ad')).toHaveLength(props.ads.length);
  });
});
