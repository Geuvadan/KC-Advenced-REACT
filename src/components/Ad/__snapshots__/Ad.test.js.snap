// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ad Snapshot testing 1`] = `
<div
  className="ads-grid"
>
  <div
    className="ad"
    key="1234567890"
  >
    <p>
      <b>
        Nombre:
      </b>
       
      test_Ad_1
    </p>
    <p>
      <b>
        Operación:
      </b>
       
      Venta
    </p>
    <p>
      <b>
        Precio:
      </b>
       
      100
      €
    </p>
    <Link
      className="button"
      to="/details/1234567890"
    >
      Más Información
    </Link>
  </div>
  <div
    className="ad"
    key="1234567890"
  >
    <p>
      <b>
        Nombre:
      </b>
       
      test_Ad_1
    </p>
    <p>
      <b>
        Operación:
      </b>
       
      Venta
    </p>
    <p>
      <b>
        Precio:
      </b>
       
      100
      €
    </p>
    <Link
      className="button"
      to="/details/1234567890"
    >
      Más Información
    </Link>
  </div>
</div>
`;
